src/Giraffe/ResponseWriters.fs

```diff
@@ -9,16 +10,47 @@ open Microsoft.Net.Http.Headers
 open FSharp.Control.Tasks.V2.ContextInsensitive
 open Giraffe.GiraffeViewEngine
 
+
 // ---------------------------
 // Helper functions
 // ---------------------------
 
+let private DefaultCapacity = 1386
+let private MaxBuilderSize  = DefaultCapacity * 3
+
+type StringBuilderCache =
+
+    [<ThreadStatic>]
+    [<DefaultValue>]
+    static val mutable private instance : StringBuilder
+
+    static member Get () = StringBuilderCache.Get(DefaultCapacity)
+    static member Get (capacity : int) =
+
+        if capacity <= MaxBuilderSize then
```

Comment:
If we always call `Get()` without a capacity (where the default capacity is used) then this if statement will always be true.